### PR TITLE
Replace Glog with Zap Logger

### PR DESCRIPTION
**Pull request by xWink**
**What type of PR is this?**

cleanup

**Which issue does this PR fix**:
#126 partially

**What does this PR do / Why do we need it**:
- Removes glog entirely
- All previous uses of glog are replaced with zap logger
- Tests that now must use an implementation of zap logger use the fallback logger
- Removal of redundant newlines in logs
- NO controller functionality changes or log content changes, this is only a change from glog to zap

**If an issue # is not available please add repro steps and logs from aws-gateway-controller showing the issue**:

**Testing done on this change**:

N/A

**Automation added to e2e**:

N/A

**Will this PR introduce any new dependencies?**:

No

**Will this break upgrades or downgrades. Has updating a running cluster been tested?**:
No. N/A

**Does this PR introduce any user-facing change?**:

Logs will be a bit cleaner, but nothing else.

```release-note

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/deploy/lattice/listener_manager.go [hunk 1]
```diff
@@ -157,7 +163,7 @@ func (s *defaultListenerManager) List(ctx context.Context, serviceID string) ([]
 }
 
 func (s *defaultListenerManager) findListenerByNamePort(ctx context.Context, serviceID string, port int64) (*vpclattice.ListenerSummary, error) {
-	glog.V(6).Infof("calling findListenerByNamePort serviceID %v port %d \n", serviceID, port)
+	s.log.Infof("calling findListenerByNamePort serviceID %v port %d", serviceID, port)
```
Comment (mikhail-aws) [inline pkg/deploy/lattice/listener_manager.go:166]: I would use debug for all logs in this method

File: pkg/deploy/lattice/listener_manager.go [hunk 2]
```diff
@@ -185,17 +191,17 @@ func (s *defaultListenerManager) findListenerByNamePort(ctx context.Context, ser
...
 	resp, err := s.cloud.Lattice().DeleteListener(&listenerDeleteInput)
 
-	glog.V(2).Infoln("############ req delete listner ###########")
-	glog.V(2).Infoln(listenerDeleteInput)
-	glog.V(2).Infoln("############resp delete listner ###########")
-	glog.V(2).Infof("Delete  listener resp %vm err :%v\n", resp, err)
+	s.log.Debugln("############ req delete listner ###########")
+	s.log.Debugln(listenerDeleteInput)
+	s.log.Debugln("############resp delete listner ###########")
+	s.log.Debugf("Delete  listener resp %vm err :%v", resp, err)
```
Comment (mikhail-aws) [inline pkg/deploy/lattice/listener_manager.go:205]: another gift

File: pkg/deploy/lattice/listener_synthesizer.go [hunk 3]
```diff
@@ -34,27 +39,27 @@ func (l *listenerSynthesizer) Synthesize(ctx context.Context) error {
...
+	l.log.Infof("Synthesize Listener:  %v", resListener)
 
 	for _, listener := range resListener {
 		status, err := l.listener.Create(ctx, listener)
 
 		if err != nil {
 			errmsg := fmt.Sprintf("ListenerSynthesie: failed to create listener %v, err %v", listener, err)
-			glog.V(6).Infof("Fail to listenerSynthesizer: %s \n", errmsg)
+			l.log.Infof("Fail to listenerSynthesizer: %s", errmsg)
 			return errors.New(errmsg)
```
Comment (mikhail-aws) [inline pkg/deploy/lattice/listener_synthesizer.go:50]: no log

File: pkg/deploy/lattice/listener_synthesizer.go [hunk 4]
```diff
@@ -34,27 +39,27 @@ func (l *listenerSynthesizer) Synthesize(ctx context.Context) error {
...
+		l.latticeStore.AddListener(listener.Spec.Name, listener.Spec.Namespace, listener.Spec.Port,
 			listener.Spec.Protocol,
 			status.ListenerARN, status.ListenerID)
 	}
 
 	// handle delete
 	sdkListeners, err := l.getSDKListeners(ctx)
 
-	glog.V(6).Infof("getSDKlistener: %v, err: %v\n", sdkListeners, err)
+	l.log.Infof("getSDKlistener: %v, err: %v", sdkListeners, err)
```
Comment (mikhail-aws) [inline pkg/deploy/lattice/listener_synthesizer.go:62]: log error with if condition

File: pkg/deploy/lattice/listener_synthesizer.go [hunk 5]
```diff
@@ -63,11 +68,11 @@ func (l *listenerSynthesizer) Synthesize(ctx context.Context) error {
 			continue
 		}
 
-		glog.V(6).Infof("ListenerSynthesize >>>> delete staled sdk listener %v\n", *sdkListener)
+		l.log.Infof("ListenerSynthesize >>>> delete staled sdk listener %v", *sdkListener)
```
Comment (mikhail-aws) [inline pkg/deploy/lattice/listener_synthesizer.go:71]: glog.V(6) is Debug level.

File: pkg/deploy/lattice/listener_synthesizer.go [hunk 6]
```diff
@@ -97,12 +102,12 @@ func (l *listenerSynthesizer) getSDKListeners(ctx context.Context) ([]*latticemo
...
-	glog.V(6).Infof("getSDKListeners service: %v err: %v \n", resService, err)
+	l.log.Infof("getSDKListeners service: %v err: %v", resService, err)
 
 	for _, service := range resService {
-		latticeService, err := l.latticestore.GetLatticeService(service.Spec.Name, service.Spec.Namespace)
+		latticeService, err := l.latticeStore.GetLatticeService(service.Spec.Name, service.Spec.Namespace)
 		if err != nil {
-			glog.V(6).Infof("getSDKRules: failed to find in store,  service: %v, err: %v \n", service, err)
+			l.log.Infof("getSDKRules: failed to find in store,  service: %v, err: %v", service, err)
 			return sdkListeners, errors.New("getSDKRules: failed to find service in store")
```
Comment (mikhail-aws) [inline pkg/deploy/lattice/listener_synthesizer.go:111]: no log, should wrap error in return

File: pkg/deploy/lattice/rule_manager.go [hunk 7]
```diff
@@ -60,11 +63,11 @@ func (r *defaultRuleManager) List(ctx context.Context, service string, listener
...
 
-	glog.V(6).Infoln("############list rules req############")
-	glog.V(6).Infof("rule: %v , serviceID: %v, listenerID %v \n", resp, service, listener)
+	r.log.Debugln("############list rules req############")
+	r.log.Debugf("rule: %v , serviceID: %v, listenerID %v", resp, service, listener)
 
-	glog.V(6).Infoln("############list rules resp############")
-	glog.V(6).Infof("resp: %v, err: %v\n", resp, err)
+	r.log.Debugln("############list rules resp############")
+	r.log.Debugf("resp: %v, err: %v", resp, err)
```
Comment (mikhail-aws) [inline pkg/deploy/lattice/rule_manager.go:70]: single log statement after err!=nil check

File: pkg/deploy/lattice/rule_manager.go [hunk 8]
```diff
@@ -90,13 +93,13 @@ func (r *defaultRuleManager) Update(ctx context.Context, rules []*latticemodel.R
 
 	var ruleUpdateList []*vpclattice.RuleUpdate
 
-	glog.V(6).Infof("Rule --- update >>>>>>>>.%v\n", rules)
+	r.log.Infof("Rule --- update >>>>>>>>.%v", rules)
```
Comment (mikhail-aws) [inline pkg/deploy/lattice/rule_manager.go:96]: debug

File: pkg/deploy/lattice/rule_manager.go [hunk 9]
```diff
@@ -90,13 +93,13 @@ func (r *defaultRuleManager) Update(ctx context.Context, rules []*latticemodel.R
...
 
 	latticeService, err := r.latticeDataStore.GetLatticeService(rules[0].Spec.ServiceName, rules[0].Spec.ServiceNamespace)
 
 	if err != nil {
 		errmsg := fmt.Sprintf("Service %v not found during rule creation", rules[0].Spec)
-		glog.V(2).Infof("Error during update rule %s \n", errmsg)
+		r.log.Debugf("Error during update rule %s", errmsg)
 		return errors.New(errmsg)
```
Comment (mikhail-aws) [inline pkg/deploy/lattice/rule_manager.go:103]: no log

File: pkg/deploy/lattice/rule_manager.go [hunk 10]
```diff
@@ -105,7 +108,7 @@ func (r *defaultRuleManager) Update(ctx context.Context, rules []*latticemodel.R
 
 	if err != nil {
 		errmsg := fmt.Sprintf("Listener %v not found during rule creation", rules[0].Spec)
-		glog.V(2).Infof("Error during update rule %s \n", errmsg)
+		r.log.Debugf("Error during update rule %s", errmsg)
 		return errors.New(errmsg)
```
Comment (mikhail-aws) [inline pkg/deploy/lattice/rule_manager.go:112]: no log

File: pkg/deploy/lattice/rule_manager.go [hunk 11]
```diff
@@ -128,22 +131,22 @@ func (r *defaultRuleManager) Update(ctx context.Context, rules []*latticemodel.R
...
 	resp, err := r.cloud.Lattice().BatchUpdateRule(&batchRuleInput)
 
-	glog.V(2).Infoln("############req updating rule ###########")
-	glog.V(2).Infoln(batchRuleInput)
-	glog.V(2).Infof("############resp updateing rule ###########, err: %v \n", err)
-	glog.V(2).Infoln(resp)
+	r.log.Debugln("############req updating rule ###########")
+	r.log.Debugln(batchRuleInput)
+	r.log.Debugf("############resp updateing rule ###########, err: %v", err)
+	r.log.Debugln(resp)
```
Comment (mikhail-aws) [inline pkg/deploy/lattice/rule_manager.go:137]: :(

File: pkg/deploy/lattice/rule_manager.go [hunk 12]
```diff
@@ -309,75 +310,75 @@ func updateSDKhttpMatch(httpMatch *vpclattice.HttpMatch, rule *latticemodel.Rule
 	}
 }
 
-func isRulesSame(modelRule *latticemodel.Rule, sdkRuleDetail *vpclattice.GetRuleOutput) bool {
+func isRulesSame(log gwlog.Logger, modelRule *latticemodel.Rule, sdkRuleDetail *vpclattice.GetRuleOutput) bool {
 	// Exact Path Match
 	if modelRule.Spec.PathMatchExact {
-		glog.V(6).Infoln("Checking PathMatchExact")
+		log.Infoln("Checking PathMatchExact")
```
Comment (mikhail-aws) [inline pkg/deploy/lattice/rule_manager.go:316]: debug and for friends below

File: pkg/deploy/lattice/rule_manager.go [hunk 13]
```diff
@@ -428,11 +429,11 @@ func (r *defaultRuleManager) findMatchingRule(ctx context.Context, rule *lattice
...
 
-	glog.V(6).Infoln("############list rules req############")
-	glog.V(6).Infof("rule: %v , serviceID: %v, listenerID %v \n", rule, serviceID, listenerID)
+	r.log.Infoln("############list rules req############")
+	r.log.Infof("rule: %v , serviceID: %v, listenerID %v", rule, serviceID, listenerID)
 
-	glog.V(6).Infoln("############list rules resp############")
-	glog.V(6).Infof("resp: %v, err: %v\n", resp, err)
+	r.log.Infoln("############list rules resp############")
+	r.log.Infof("resp: %v, err: %v", resp, err)
```
Comment (mikhail-aws) [inline pkg/deploy/lattice/rule_manager.go:436]: single log after error hanlding

File: pkg/deploy/lattice/rule_manager.go [hunk 14]
```diff
@@ -460,13 +461,13 @@ func (r *defaultRuleManager) findMatchingRule(ctx context.Context, rule *lattice
 		ruleResp, err := r.cloud.Lattice().GetRule(&ruleInput)
 
 		if err != nil {
-			glog.V(2).Infof("findMatchingRule, rule %v not found err:%v\n", ruleInput, err)
+			r.log.Debugf("findMatchingRule, rule %v not found err:%v", ruleInput, err)
```
Comment (mikhail-aws) [inline pkg/deploy/lattice/rule_manager.go:464]: v2 is info, v6 is debug

File: pkg/deploy/lattice/rule_manager.go [hunk 15]
```diff
@@ -579,7 +580,7 @@ func (r *defaultRuleManager) Delete(ctx context.Context, ruleID string, listener
 
 	resp, err := r.cloud.Lattice().DeleteRule(&deleteInput)
 
-	glog.V(2).Infof("Delete Rule >>>> input %v, output %v, err %v\n", deleteInput, resp, err)
+	r.log.Debugf("Delete Rule >>>> input %v, output %v, err %v", deleteInput, resp, err)
 
 	return err
```
Comment (mikhail-aws) [inline pkg/deploy/lattice/rule_manager.go:585]: no log, should wrap error with details from this log

File: pkg/deploy/lattice/rule_synthesizer.go [hunk 16]
```diff
@@ -29,28 +36,28 @@ func (r *ruleSynthesizer) Synthesize(ctx context.Context) error {
 
 	err := r.stack.ListResources(&resRule)
 
-	glog.V(6).Infof("Synthesize rule = %v, err :%v \n", resRule, err)
+	r.log.Infof("Synthesize rule = %v, err :%v", resRule, err)
```
Comment (mikhail-aws) [inline pkg/deploy/lattice/rule_synthesizer.go:39]: if err, then log error

File: pkg/deploy/lattice/rule_synthesizer.go [hunk 17]
```diff
@@ -29,28 +36,28 @@ func (r *ruleSynthesizer) Synthesize(ctx context.Context) error {
...
 	updatePriority := false
 
 	for _, rule := range resRule {
 		ruleResp, err := r.rule.Create(ctx, rule)
 
 		if err != nil {
-			glog.V(6).Infof("Failed to create rule %v, err :%v \n", rule, err)
+			r.log.Infof("Failed to create rule %v, err :%v", rule, err)
 			return err
```
Comment (mikhail-aws) [inline pkg/deploy/lattice/rule_synthesizer.go:47]: no log on error return

File: pkg/deploy/lattice/rule_synthesizer.go [hunk 18]
```diff
@@ -29,28 +36,28 @@ func (r *ruleSynthesizer) Synthesize(ctx context.Context) error {
...
 
-		glog.V(6).Infof("Synthesise rule %v, ruleResp:%v \n", rule, ruleResp)
+		r.log.Infof("Synthesise rule %v, ruleResp:%v", rule, ruleResp)
 		rule.Status = &ruleResp
 	}
 
 	// handle delete
 	sdkRules, err := r.getSDKRules(ctx)
-	glog.V(6).Infof("rule>>> synthesize,  sdkRules :%v err: %v \n", sdkRules, err)
+	r.log.Infof("rule>>> synthesize,  sdkRules :%v err: %v", sdkRules, err)
```
Comment (mikhail-aws) [inline pkg/deploy/lattice/rule_synthesizer.go:60]: if err, then log

File: pkg/deploy/lattice/rule_synthesizer.go [hunk 19]
```diff
@@ -76,32 +83,32 @@ func (r *ruleSynthesizer) findMatchedRule(ctx context.Context, sdkRuleID string,
...
-		glog.V(6).Infof("findMatchRule, rule not found err:%v\n", err)
+		r.log.Infof("findMatchRule, rule not found err:%v", err)
 		return modelRule, errors.New("rule not found")
 	}
 
 	if sdkRuleDetail.Match == nil ||
 		sdkRuleDetail.Match.HttpMatch == nil {
-		glog.V(6).Infof("no HTTPMatch ")
+		r.log.Infof("no HTTPMatch ")
 		return modelRule, errors.New("rule not found")
```
Comment (mikhail-aws) [inline pkg/deploy/lattice/rule_synthesizer.go:97]: no log

**Review by mikhail-aws — CHANGES_REQUESTED** [review]
Reviewed half of it. Before going further few major comments.

1. Approach, lets agree on the that. I would expect if we touch code, we dont want to come back again for another logging improvement. So it's better to have a fewer but complete changes, than many and come back again. Why come back again, will list below.
2. glog.V6 is debug, there are many places where debug converted into info log.
3. glog.v2 is info, there are places where info converted into debug.
4. multi-line logging - merge into one
5. log error or return error, dont do both
6. statements with `foo,err := getFoo()`, and then we log both log.Info(foo, err), not good, if error happened and we handle it by logging, we do if err != nil { log(err) }
7. Info logging with %v - we talked about it - no %v, log only fields that we care about. Not sure which fields are important, log just name or id. For debugging can be used, carefully.

**Comment by solmonk** [comment]
In addition - let's remove "#####" or ">>>>>" style logs whenever possible (sorry not sure how to call these)

This could be too much for one PR so we may just split the PR by components - (do controller, do manager, do modelbuild, and so on)

**Comment by solmonk** [comment]
btw I'm totally happy with replacing a few info logs with debug logs by good judgement. I would like to see minimal set of logs when debug mode is off

**Comment by xWink** [comment]
Going to close this and use it as reference for fresh, smaller PRs.

First one is up
https://github.com/aws/aws-application-networking-k8s/pull/410

**Pull request closed without merging**